source/Cosmos.System2/Network/IPv4/TCP/Tcp.cs

```diff
@@ -0,0 +1,528 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Cosmos.HAL;
+using Cosmos.System.Network.Config;
+
+namespace Cosmos.System.Network.IPv4.TCP
+{
+    /// <summary>
+    /// TCP Connection status
+    /// </summary>
+    public enum Status
+    {
+        LISTEN,
+        SYN_SENT,
+        SYN_RECEIVED,
+        ESTABLISHED,
+        FIN_WAIT1,
+        FIN_WAIT2,
+        CLOSE_WAIT,
+        CLOSING,
+        LAST_ACK,
+        TIME_WAIT,
+        CLOSED
+    }
+
+    /// <summary>
+    /// Tcp class. Used to manage the TCP state machine
```

Comment:
Could you expand on the comment a bit more about how it is used by the rest of the code, so its easier for someone reading the code to understand the class in the context of the network stack.